### PR TITLE
RAM - implement CRUD endpoints

### PR DESCRIPTION
**Pull request by gruebel**
It is a basic implementation of the AWS Resource Access Manager plus an endpoint for enabling sharing with AWS Organizations.

A lot of functionality is missing, so feel free to open up some issues and when I have time, I will implement them :)

### CODE REVIEW

File: moto/ram/models.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    def update(self, **kwargs):
+        self.allow_external_principals = kwargs.get(
+            "allowExternalPrincipals", self.allow_external_principals
+        )
+        self.last_updated_time = datetime.utcnow()
+        self.name = kwargs.get("name", self.name)
+
+    def _create_arn(self):
+        id = "{0}-{1}-{2}-{3}-{4}".format(
```
Comment (bblommers) [inline moto/ram/models.py:152]: This looks like a UUID - does it make sense to that instead? (`from uuid import uuid4`?)

File: tests/test_ram/test_ram.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    # when
+    response = client.create_resource_share(name="test")
+
+    # then
+    resource = response["resourceShare"]
+    resource["allowExternalPrincipals"].should.be.ok
+    resource["creationTime"].should.be.a(datetime)
+    resource["lastUpdatedTime"].should.be.a(datetime)
+    resource["name"].should.equal("test")
+    resource["owningAccountId"].should.equal("123456789012")
```
Comment (bblommers) [inline tests/test_ram/test_ram.py:26]: You can import the ACCOUNT_ID from `moto.core`, no need to hardcode it everytime

File: moto/ram/models.py (hunk 1 above)
Comment (gruebel) [reply moto/ram/models.py]: Good eye! Somehow didn't realize it 😄

File: tests/test_ram/test_ram.py (hunk 2 above)
Comment (gruebel) [reply tests/test_ram/test_ram.py]: Good point, I got used to hardcode the results of my tests.

**Comment by bblommers** [comment]
This is now part of moto >= 1.3.15.dev980

**Comment by kmcquade** [comment]
@bblommers @gruebel - thought I'd point out - RAM support is not noted in the README

**Comment by bblommers** [comment]
Thanks @kmcquade - I've added it now

**Pull request merged**